src/com/android/settings/wifi/WifiTimeoutSettings.java

```diff
@@ -0,0 +1,142 @@
+import android.content.pm.PackageManager;
+import android.graphics.drawable.Drawable;
+import android.provider.Settings;
+import android.text.format.DateUtils;
+
+import androidx.preference.PreferenceScreen;
+
+import com.android.settings.R;
+import com.android.settings.search.BaseSearchIndexProvider;
+import com.android.settings.widget.RadioButtonPickerFragment;
+import com.android.settingslib.search.SearchIndexable;
+import com.android.settingslib.widget.CandidateInfo;
+import com.android.settingslib.widget.RadioButtonPreference;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@SearchIndexable(forTarget = SearchIndexable.ALL & ~SearchIndexable.ARC)
+public class WifiTimeoutSettings extends RadioButtonPickerFragment {
+    public static final BaseSearchIndexProvider SEARCH_INDEX_DATA_PROVIDER =
+            new BaseSearchIndexProvider(R.xml.wifi_configure_settings) {
+                @Override
+                protected boolean isPageSearchEnabled(Context context) {
+                    return context.getPackageManager().hasSystemFeature(
+                            PackageManager.FEATURE_WIFI);
+                }
+            };
+    static final long[] CANDIDATES = {
+            0,
+            TimeUnit.SECONDS.toMillis(15),
+            TimeUnit.SECONDS.toMillis(30),
+            TimeUnit.MINUTES.toMillis(1),
+            TimeUnit.MINUTES.toMillis(2),
+            TimeUnit.MINUTES.toMillis(5),
+            TimeUnit.MINUTES.toMillis(10),
+            TimeUnit.MINUTES.toMillis(30),
+            TimeUnit.HOURS.toMillis(1),
+            TimeUnit.HOURS.toMillis(2),
+            TimeUnit.HOURS.toMillis(4),
+            TimeUnit.HOURS.toMillis(8),
+    };
+
+    public static String toHumanFriendly(long duration, Context context) {
+        if (duration == 0) {
+            return context.getString(R.string.wifi_timeout_summary_never);
+        }
+        return DateUtils.formatDuration(duration).toString();
+    }
+
+    @Override
+    protected List<? extends CandidateInfo> getCandidates() {
+        final List<CandidateInfo> candidates = new ArrayList<>();
+
+        for (long timeout : CANDIDATES) {
+            candidates.add(
+                    new TimeoutCandidateInfo(
+                            toHumanFriendly(timeout, getContext()),
+                            String.valueOf(timeout)
+                    )
+            );
+        }
+        return candidates;
+    }
+
+    @Override
+    public void updateCandidates() {
+        final PreferenceScreen screen = getPreferenceScreen();
+        screen.removeAll();
+
+        final List<? extends CandidateInfo> candidateList = getCandidates();
+        if (candidateList == null) {
+            return;
+        }
+
+        for (CandidateInfo info : candidateList) {
+            RadioButtonPreference pref = new RadioButtonPreference(getPrefContext());
+            bindPreference(pref, info.getKey(), info, getDefaultKey());
```

Comment:
`getDefaultKey()` calls a relatively expensive `Settings.Global.getLong` method.
Calling it for each iteration of the loop is not necessary, it can be called once before the loop begins.